test/unit/test_model.py

```diff
@@ -89,6 +89,152 @@ def test_set_cpu_feature_nx(self):
         except:
             assert False
 
+    def test_set_ahci_storage_controller(self):
+        try:
+            backend_storage_info = [{
+                "controller": {
+                    "type": "ahci",
+                    "max_drive_per_controller": 6,
+                    "drives": [{"size": 8}]
+                }
+            }]
+            storage = model.CBackendStorage(backend_storage_info)
+            storage.init()
+            storage.precheck()
+            storage.handle_parms()
+            assert "-device ahci" in storage.get_option()
+        except:
+            assert False
+
+    def test_set_scsi_storage_controller(self):
+        try:
+            backend_storage_info = [{
+                "controller": {
+                    "type": "scsi",
+                    "max_drive_per_controller": 8,
+                    "drives": [{"size": 8}]
+                }
+            }]
+            storage = model.CBackendStorage(backend_storage_info)
+            storage.init()
+            storage.precheck()
+            storage.handle_parms()
+            assert "-device scsi" in storage.get_option()
+        except:
+            assert False
+
+    def test_set_ahci_storage_controller_2x(self):
+        try:
+            backend_storage_info = [{
+                "controller": {
+                    "type": "ahci",
+                    "max_drive_per_controller": 2,
+                    "drives": [{"size": 8}, {"size": 8}, {"size": 8}]
```

Comment:
If the drive number exceed the max_drive_per_controller, it will automatically add a new controller to make drives enabled